wodles/aws/aws_s3.py

```diff
@@ -3325,7 +3306,189 @@ def purge_db(self, log_group):
+
+        Parameters
+        ----------
+        message_body : dict
+            An SQS message received from the queue.
+        """
+        raise NotImplementedError
+
+
+class AWSSubscriberBucket(WazuhIntegration, AWSS3LogHandler):
+    """
+    Class for processing events from AWS S3 buckets.
+
+    Attributes
+    ----------
+    aws_profile : str
+        AWS profile.
+    iam_role_arn : str
+        IAM Role.
+    """
+    def __init__(self, service_endpoint: str = None, sts_endpoint: str = None, profile: str = None, **kwargs):
+        WazuhIntegration.__init__(self, access_key=None,
+                                  secret_key=None,
+                                  aws_profile=profile,
+                                  service_name='s3',
+                                  service_endpoint=service_endpoint,
+                                  sts_endpoint=sts_endpoint,
+                                  **kwargs)
+
+    def obtain_logs(self, bucket: str, log_path: str) -> list[dict]:
+        """
+        Fetch a file from a bucket and obtain a list of events from it.
+
+        Parameters
+        ----------
+        bucket : str
+            Bucket to get the file from.
+        log_path : str
+            Relative path of the file inside the bucket.
+
+        Returns
+        -------
+        list[dict]
+            List of extracted events to send to Wazuh.
+        """
+
+        def _process_jsonl(file):
+            json_list = list(file)
+            result = []
+            for json_item in json_list:
+                x = json.loads(json_item)
+                result.append(dict(x, integration='aws'))
+            return result
+
+        def json_event_generator(data):
+            while data:
+                json_data, json_index = decoder.raw_decode(data)
+                data = data[json_index:]
+                yield json_data
+
+        with self.decompress_file(bucket, log_key=log_path) as f:
+            try:
+                if log_path.endswith('.jsonl.gz'):
+                    return _process_jsonl(f)
+
+                decoder = json.JSONDecoder()
+                return [dict(event.get('detail', event), source="custom")
+                        for event in json_event_generator(f.read())]
+
+            except json.JSONDecodeError:
+                debug(f"+++ Log file does not contain JSON objects. Trying with csv format.", 2)
+                f.seek(0)
+                try:
+                    dialect = csv.Sniffer().sniff(f.read(1024))
+                    f.seek(0)
+                    reader = csv.DictReader(f, dialect=dialect)
+                    return [dict({k: v for k, v in row.items() if v is not None},
+                                 source='custom') for row in reader]
+                except csv.Error:
+                    print(f"ERROR: Data in the file does not seem to be CSV either.")
+                    sys.exit(9)
```

Comment:
Does a use case to bring plain logs (neither JSON nor CSV) make sense? And if it does, it's not currently supported, is it?

Reply:
It is not currently supported. We could analyze it since users could be redirecting plain text logs to their buckets. 
We could send plain text logs in the following format:
```
{
  "integration": "aws", 
  "aws": {
      "full_log": <plain_text_log>
   }
} 
```
We could accept multiple line files, having the logs separated by a line feed (`\n`), and the user would have to define their own decoders for the contents of the `full_log` field

Reply:
Done, execution example [here](https://github.com/wazuh/wazuh/issues/13577#issuecomment-1662919867).